packages/next/src/server-build.ts

```diff
@@ -1288,32 +1315,39 @@ export async function serverBuild({
     hasPages404: routesManifest.pages404,
     isCorrectNotFoundRoutes,
     isEmptyAllowQueryForPrendered,
+    omittedPrerenderRoutes,
   });
 
-  await Promise.all(
-    Object.keys(prerenderManifest.staticRoutes).map(route =>
+  const prerenderPromises: Array<Promise<void>> = [];
```

Comment:
We can't use these prerender promises as it causes a race condition with any of them rejecting before we attach the `Promise.all` which crashes the build.